akka-docs/src/main/paradox/typed/persistence-durable-state.md

```diff
@@ -0,0 +1,391 @@
+
+Java
+:  @@snip [DurableStatePersistentBehaviorTest.java](/akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/DurableStatePersistentBehaviorTest.java) { #effects }
+
+Most of the time this will be done with the `thenRun` method on the `Effect` above. You can factor out
+common side effects into functions and reuse for several commands. For example:
+
+Scala
+:  @@snip [PersistentActorCompileOnlyTest.scala](/akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/PersistentActorCompileOnlyTest.scala) { #commonChainedEffects }
+
+Java
+:  @@snip [PersistentActorCompileOnlyTest.java](/akka-persistence-typed/src/test/java/akka/persistence/typed/javadsl/PersistentActorCompileOnlyTest.java) { #commonChainedEffects }
+
+### Side effects ordering and guarantees
+
+Any side effects are executed on an at-most-once basis and will not be executed if the persist fails.
+
+Side effects are not run when the actor is restarted or started again after being stopped.
+
+The side effects are executed sequentially, it is not possible to execute side effects in parallel, unless they
+call out to something that is running concurrently (for example sending a message to another actor).
+
+It's possible to execute a side effect before persisting the state, but that can result in that the
+side effect is performed but the state is not stored if the persist fails.
+
+## Cluster Sharding and DurableStateBehavior
+
+In a use case where the number of persistent actors needed is higher than what would fit in the memory of one node or
+where resilience is important so that if a node crashes the persistent actors are quickly started on a new node and can
+resume operations @ref:[Cluster Sharding](cluster-sharding.md) is an excellent fit to spread persistent actors over a
+cluster and address them by id.
+
+The `DurableStateBehavior` can then be run as with any plain actor as described in @ref:[actors documentation](actors.md),
+but since Akka Persistence is based on the single-writer principle the persistent actors are typically used together
+with Cluster Sharding. For a particular `persistenceId` only one persistent actor instance should be active at one time.
+Cluster Sharding ensures that there is only one active entity for each id. 
+
+## Accessing the ActorContext
+
+If the @apidoc[DurableStateBehavior] needs to use the @apidoc[typed.*.ActorContext], for example to spawn child actors, it can be obtained by
+wrapping construction with `Behaviors.setup`:
+
+Scala
+:  @@snip [DurableStatePersistentBehaviorCompileOnly.scala](/akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/DurableStatePersistentBehaviorCompileOnly.scala) { #actor-context }
+
+Java
+:  @@snip [BasicPersistentBehaviorTest.java](/akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/DurableStatePersistentBehaviorTest.java) { #actor-context }
+
+## Changing Behavior
+
+After processing a message, actors are able to return the `Behavior` that is used
```

Comment:
Maybe simpler to just state something like : Unlike regular actors, persistent actors cannot return a different behavior for a next message.

Reply:
I would prefer it if this is written, focused on how it should be done, and then a note saying "while reading this you might have wondered why we did not use ..., this is not allowed because' or something and explaining there why the behavior changing cannot work with akka persistence.

Reply:
refactored ..